### PR TITLE
Hide 'your passwords don't match' for invalid passwords

### PR DESCRIPTION
**Pull request by adam-incuna**
@incuna/backend review please?

This is a fix for version 8.1.0, for a project that doesn't want to update to DRF 3 or any of that complicated stuff.  Please don't merge (even if GitHub lets you, which it hopefully won't :P).  I'll release from this branch when review is done.

### CODE REVIEW

File: user_management/api/tests/test_serializers.py [hunk 1]
```diff
@@ -161,7 +162,8 @@ def test_deserialize_mismatched_passwords(self):
         self.data['password2'] = 'different_password'
         serializer = serializers.RegistrationSerializer(data=self.data)
         self.assertFalse(serializer.is_valid())
-        self.assertIn('password2', serializer.errors)
+        message = 'Your passwords do not match.'
+        self.assertIn(message, serializer.errors['non_field_errors'])
```
Comment (Ian-Foote) [inline user_management/api/tests/test_serializers.py]: What is `serializer.errors['non_field_errors']` here? If it's `'Your passwords do not match.'`, this should be `self.assertEqual`.

File: user_management/api/tests/test_serializers.py [hunk 2]
```diff
@@ -75,6 +75,7 @@ def test_deserialize_invalid_new_password(self):
         })
         self.assertFalse(serializer.is_valid())
         self.assertIn('new_password', serializer.errors)
+        self.assertNotIn('Your passwords do not match', serializer.errors)
```
Comment (Ian-Foote) [inline user_management/api/tests/test_serializers.py]: `Your passwords do not match` is a value, not a key.

Also, shouldn't it end in a `.`?

File: user_management/api/tests/test_serializers.py (hunk 1 above)
Comment (adam-incuna) [inline user_management/api/tests/test_serializers.py]: It's a list of that message, yes.  I'm keen on allowing this test to be generic though - we don't care that there aren't _more_ errors in `non_field_errors`, only that this one is present.

File: user_management/api/tests/test_serializers.py (hunk 2 above)
Comment (adam-incuna) [inline user_management/api/tests/test_serializers.py]: Yup, missed that. Thanks!

**Comment by adam-incuna** [comment]
Updated.

File: user_management/api/tests/test_serializers.py (hunk 1 above)
Comment (Ian-Foote) [inline user_management/api/tests/test_serializers.py]: Ok.

**Comment by Ian-Foote** [comment]
LGTM, but I notice travis is failing. Do the tests pass locally?

**Comment by adam-incuna** [comment]
Yup!

**Comment by adam-incuna** [comment]
(And thanks :) )

**Comment by adam-incuna** [comment]
screen shot 2015-08-04 at 17 36 38

**Comment by meshy** [comment]
Looks like the build env has changed in such a way as to break `colour-runner` `:/`

**Comment by adam-incuna** [comment]
Is this mergeable?

**Comment by adam-incuna** [comment]
Hmm what

**Comment by adam-incuna** [comment]
That's really weird.  It works fine locally.  Help? :/

**Comment by meshy** [comment]
I think you need to email travis

**Comment by adam-incuna** [comment]
Um.  What exactly do I say to them?  I don't know what's going on here.

**Comment by Ian-Foote** [comment]
If I understand this issue correctly, we can just tag and push the last commit here and make a bugfix release. After that we can merge master to fix the conflicts allowing us to merge back to master to update the changelog.

I don't think we should worry too much about travis failing at this old location in the code base, since the tests pass locally.

**Comment by adam-incuna** [comment]
@Ian-Foote good thinking!  Will do all that now.

**Comment by adam-incuna** [comment]
Dumb question - how do I deploy DUM?  It doesn't seem to have a `make release` or similar.

**Comment by meshy** [comment]
@adam-incuna is there any reason why this should not be merged into master?

**Comment by adam-incuna** [comment]
@meshy It's already been done in master, I believe.  In fact I'm pretty sure we can close this issue now.

**Comment by meshy** [comment]
Oh it's already there?! Fantastic.

Any chance of updating the CHANGELOG with this odd version?

**Comment by adam-incuna** [comment]
Yeah, I should do that, shouldn't I.

**Comment by meshy** [comment]
Ta. (I need to do another 8.1.x fix)

**Comment by adam-incuna** [comment]
https://github.com/incuna/django-user-management/pull/131

**Pull request closed without merging**